core/src/mindustry/entities/comp/UnitComp.java

```diff
@@ -214,6 +214,8 @@ public double sense(LAccess sensor){
             case ammoCapacity -> type.ammoCapacity;
             case x -> World.conv(x);
             case y -> World.conv(y);
+            case velocityX -> vel.x;
+            case velocityY -> vel.y;
```

Comment:
This returns velocity in world units, but logic operates in tiles. It must be converted, like speed.